checkstyle/import-control.xml

```diff
@@ -98,6 +98,8 @@
       <allow pkg="org.apache.kafka.common.protocol" />
       <allow pkg="org.apache.kafka.common.protocol.types" />
       <allow pkg="org.apache.kafka.common.errors" />
+      <!-- MemoryRecordsBuilder needs to access TransactionState, which seems to naturally belong in the Producer -->
+      <allow pkg="org.apache.kafka.clients.producer" />
```

Comment:
Yes. That is awkward. The dependence on the new `TransactionState` class is very minimal. I refactored to remove the dependency: https://github.com/confluentinc/kafka/pull/143